packages/amplify_core/lib/src/types/notifications/push/push_notification_message.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:amplify_core/amplify_core.dart';
+
+/// {@template amplify_core.push.push_notification_message}
+/// The Push Notification message that represents both the FCM and APNS Native notification types.
+/// {@endtemplate}
+class PushNotificationMessage
+    with AWSDebuggable, AWSSerializable<Map<String, Object?>> {
+  /// {@macro amplify_core.push.push_notification_message}
+  PushNotificationMessage({
```

Comment:
`const` with `final` fields

Reply:
Can't make it final since value may be assigned in fromJson function.

Reply:
It can. `fromJson` should be a factory.